### PR TITLE
Merge dev giddh 2.0

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.ts [hunk 1]
```diff
@@ -128,27 +127,7 @@ export class AdvanceSearchModelComponent implements OnInit, OnDestroy, OnChanges
             }
         })).pipe(takeUntil(this.destroyed$));
 
-        // Get groups with accounts
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/advance-search/advance-search.component.ts:131]: This must not be deleted, this is updated code.

**Review by aditi-tiwari17 — APPROVED** [review]

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by shalinee-walkover — APPROVED** [review]

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 2]
```diff
@@ -5095,6 +5540,147 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+    }
+
+    /**
+     * Search query change handler
+     *
+     * @param {string} query Search query
+     * @param {number} [page=1] Page to request
+     * @memberof ProformaInvoiceComponent
+     */
+    public onSearchQueryChanged(query: string, page: number = 1, searchType: string) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:5562]: Add return type

File: apps/web-giddh/src/app/theme/sales-ng-virtual-select/sh-select.component.ts [hunk 3]
```diff
@@ -225,11 +241,11 @@ export class SalesShSelectComponent implements ControlValueAccessor, OnInit, Aft
         }
         this.clearFilter();
 
-        if (!this.multiple) {
-            if (this._selectedValues[0] && this._selectedValues[0].value === item.value) {
-                callChanges = false;
-            }
-        }
+        // if (!this.multiple) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/theme/sales-ng-virtual-select/sh-select.component.ts:244]: remove commented code if not required

File: apps/web-giddh/src/app/theme/ng-virtual-select/sh-select.component.ts [hunk 4]
```diff
@@ -233,11 +263,11 @@ export class ShSelectComponent implements ControlValueAccessor, OnInit, AfterVie
         }
         this.clearFilter();
 
-        if (!this.multiple) {
-            if (this._selectedValues[0] && this._selectedValues[0].value === item.value) {
-                callChanges = false;
-            }
-        }
+        // if (!this.multiple) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/theme/ng-virtual-select/sh-select.component.ts:266]: remove commented code if not required

**Review by rohit-walkover — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**